### PR TITLE
fixes #3260 by making cached split info a softref

### LINKED ISSUE #3260: Split computation caching could cause memory usage problems in the tserver
**Describe the bug**

The changes in #3249 introduce per tablet caching of split information so that it does not need to be recomputed.  If a tablet happens to have large keys and those end up getting cached by the new code in #3249 then it could cause memory pressure or OOMEs in the tablet server.

**Expected behavior**

Computing tablet split points for tablets with large keys does not increase tablet server memory pressure after the computation.

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 1]
```diff
@@ -1821,7 +1822,8 @@ private SplitComputations(Set<FileRef> inputFiles, SortedMap<Double,Key> midPoin
     }
   }
 
-  private AtomicReference<SplitComputations> lastSplitComputation = new AtomicReference<>();
+  private volatile SoftReference<SplitComputations> lastSplitComputation =
+      new SoftReference<>(null);
```
Comment (dlmarion) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java:1826]: the lock below should be `final` ?

**Review by dlmarion — APPROVED** [review]
I think this looks good, one minor comment

**Comment by ctubbsii** [comment]
> _ No description provided. _

This commit warrants a description when this gets merged.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java (hunk 1 above)
Comment (keith-turner) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java:1826]: changed in [682832a](https://github.com/apache/accumulo/pull/3261/commits/682832a88b58f109b60aae38edd3ff215b7b6708)

**Pull request merged**